### PR TITLE
EOF Differential Layout Fuzzer

### PR DESCRIPTION
**Pull request by shemnon**
## PR description

EOF Differential Layout Fuzzer guided by Besu's layout parser.

### Thanks for sending a pull request! Have you done the following?

- [ ] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [ ] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [ ] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [ ] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [ ] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: testfuzz/build.gradle [hunk 1]
```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright ConsenSys AG.
```
Comment (macfarla) [inline testfuzz/build.gradle:2]:
```suggestion
 * Copyright contributors to Hyperledger Besu.
```

File: testfuzz/build.gradle [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  mainClass = 'org.hyperledger.besu.testfuzz.BesuFuzz'
+  args = [
+    "eof-container",
+    "--tests-dir=${projectDir}/../ethereum/referencetests/src/reference-test/external-resources/EOFTests",
+    "--corpus-dir=${buildDir}/generated/corpus",
+    "--client=evm1=evmone-eofparse",
+    "--client=revm=revme bytecode",
+    // "--client=neth=netheofparse -x",
+    // "--client=geth=eofdump eofparser",
+    // "--client=etjs=tsx ../../../ethereumjs/ethereumjs-monorepo/packages/evm/scripts/eofContainerValidator.ts"
```
Comment (macfarla) [inline testfuzz/build.gradle:67]: are these comments left in for a reason

File: testfuzz/src/main/java/org/hyperledger/besu/testfuzz/BesuFuzz.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright ConsenSys AG.
```
Comment (macfarla) [inline testfuzz/src/main/java/org/hyperledger/besu/testfuzz/BesuFuzz.java:2]:
```suggestion
 * Copyright contributors to Hyperledger Besu.
```

File: testfuzz/src/main/java/org/hyperledger/besu/testfuzz/BesuFuzzCommand.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright ConsenSys AG.
```
Comment (macfarla) [inline testfuzz/src/main/java/org/hyperledger/besu/testfuzz/BesuFuzzCommand.java:2]:
```suggestion
 * Copyright contributors to Hyperledger Besu.
```

File: testfuzz/src/main/java/org/hyperledger/besu/testfuzz/VersionProvider.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright ConsenSys AG.
```
Comment (macfarla) [inline testfuzz/src/main/java/org/hyperledger/besu/testfuzz/VersionProvider.java:2]:
```suggestion
 * Copyright contributors to Hyperledger Besu.
```

File: build.gradle [hunk 6]
```diff
@@ -160,6 +160,10 @@ allprojects {
       url 'https://splunk.jfrog.io/splunk/ext-releases-local'
       content { includeGroupByRegex('com\\.splunk\\..*') }
     }
+    maven {
+      url 'https://gitlab.com/api/v4/projects/19871573/packages/maven'
```
Comment (fab-10) [inline build.gradle:164]: is this URL stable?

**Review by fab-10 — COMMENTED** [review]
few little things and is there already a doc on how to use it?

File: build.gradle (hunk 6 above)
Comment (shemnon) [reply build.gradle:164]: It's [the documented source to use.](https://gitlab.com/gitlab-org/security-products/analyzers/fuzzers/javafuzz#installing), so it it advertised as stable.

File: testfuzz/build.gradle (hunk 2 above)
Comment (shemnon) [reply testfuzz/build.gradle]: I re-worked this section, so closing this comment as moot.

File: testfuzz/src/main/java/org/hyperledger/besu/testfuzz/BesuFuzzCommand.java [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+ * execute this command.
+ *
+ * <p>The class provides various options for setting up and executing EVM transactions. These
+ * options include, but are not limited to, setting the gas price, sender address, receiver address,
+ * and the data to be sent with the transaction.
+ *
+ * <p>Key methods in this class include 'run()' for executing the command, 'execute()' for setting
+ * up and running the EVM transaction, and 'dumpWorldState()' for outputting the current state of
+ * the Ethereum world state.
+ */
```
[the remark this answers is not in the document]
Comment (shemnon) [reply testfuzz/src/main/java/org/hyperledger/besu/testfuzz/BesuFuzzCommand.java:41]: copypasta

File: testfuzz/src/main/java/org/hyperledger/besu/testfuzz/BesuFuzz.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+package org.hyperledger.besu.testfuzz;
+
+import org.hyperledger.besu.util.LogConfigurator;
+
+/** The main entry point for the EVM (Ethereum Virtual Machine) tool. */
+public final class BesuFuzz {
+
+  /** Default constructor for the EvmTool class. */
+  public BesuFuzz() {
+    // this is here only for javaodc linting
```
[the remark this answers is not in the document]
Comment (shemnon) [reply testfuzz/src/main/java/org/hyperledger/besu/testfuzz/BesuFuzz.java:24]: Fixed.

**Review by shemnon — COMMENTED** [review]
I added a README.md.  I don't think more docs are needed as there is an audience of maybe 10-20 people.

**Review by fab-10 — APPROVED** [review]

**Pull request merged**